lib/reporting/reports/customers/base.rb

```diff
@@ -5,28 +5,56 @@ module Reports
     module Customers
       class Base < ReportTemplate
         def query_result
-          filter Spree::Order.managed_by(@user)
+          filter(Spree::Order.managed_by(@user)
             .distributed_by_user(@user)
             .complete.not_state(:canceled)
-            .order(:id)
+            .order(:id))
+            .group_by do |order|
+            {
+              customer_id: order.customer_id,
```

Comment:
It's possible to delete a customer record and then the customer id is null. This would group all deleted customers under one even though they were separate people. We could group by either customer id or email. Maybe not 100% correct in the long term because email addresses can be re-used by other people but we could do this:
```suggestion
              customer_id: order.customer_id || order.email,
              hub_id: order.distributor_id,
```

I added the hub_id in case the customer_id is not present. Otherwise it doesn't hurt because each customer id has exactly one distributor id.

Reply:
Oh nice catch, thanks a lot. Makes sense!!